tests/vmi_networking_test.go

```diff
@@ -118,43 +118,53 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	}
 
-	testMultipleVirtualMachinesConnectivityUsingDefaultBindingMechanism := func() {
-		Describe("Multiple virtual machines connectivity using bridge binding interface", func() {
+	setDefaultBindingMechanism := func(bindingMechanism v1.NetworkInterfaceType) {
+		tests.UpdateClusterConfigValueAndWait("default-network-interface", string(bindingMechanism))
+	}
+
+	setDefaultNetworkExplicitly := func(vmi *v1.VirtualMachineInstance, bindingMechanism v1.NetworkInterfaceType) {
+		switch bindingMechanism {
+		case v1.BridgeInterface:
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		case v1.MasqueradeInterface:
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		default:
+			panic(fmt.Sprintf("Selected binding %q is invalid", bindingMechanism))
+		}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+	}
+
+	testMultipleVirtualMachinesConnectivityUsingDefaultBindingMechanism := func(bindingMechanism v1.NetworkInterfaceType) {
+		Describe(fmt.Sprintf("Multiple virtual machines connectivity using %s binding mechanism", bindingMechanism), func() {
 			tests.BeforeAll(func() {
 				tests.BeforeTestCleanup()
 
+				setDefaultBindingMechanism(bindingMechanism)
+
 				// Prepare inbound and outbound VMI definitions
 
 				// inboundVMI expects implicitly to be added to the pod network
 				inboundVMI = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.ContainerDiskFor(tests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
 				inboundVMI.Labels = map[string]string{"expose": "me"}
 				inboundVMI.Spec.Subdomain = "myvmi"
 				inboundVMI.Spec.Hostname = "my-subdomain"
-				// Remove the masquerade interface to use the default bridge one
+				// Remove the masquerade interface to use the default one
 				inboundVMI.Spec.Domain.Devices.Interfaces = nil
 				inboundVMI.Spec.Networks = nil
 
 				// outboundVMI is used to connect to other vms
 				outboundVMI = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.ContainerDiskFor(tests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-				// Remove the masquerade interface to use the default bridge one
+				// Remove the masquerade interface to use the default one
 				outboundVMI.Spec.Domain.Devices.Interfaces = nil
 				outboundVMI.Spec.Networks = nil
 
 				// inboudnVMIWithPodNetworkSet adds itself in an explicit fashion to the pod network
 				inboundVMIWithPodNetworkSet = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.ContainerDiskFor(tests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-				// Remove the masquerade interface to use the default bridge one
-				inboundVMIWithPodNetworkSet.Spec.Domain.Devices.Interfaces = nil
-				inboundVMIWithPodNetworkSet.Spec.Networks = nil
-				v1.SetDefaults_NetworkInterface(inboundVMIWithPodNetworkSet)
-				Expect(inboundVMIWithPodNetworkSet.Spec.Domain.Devices.Interfaces).NotTo(BeEmpty())
```

Comment:
in the original (left side) we set the `inboundVMIWithPodNetworkSet` with [SetDefaults_NetworkInterface](https://github.com/kubevirt/client-go/blob/e297527e7722c2a1aa3445dfc0b77bac29504ac5/api/v1/defaults.go#L145), then `Expect` it to not be empty, 
why did we remove this?

Reply:
I know it is not empty since it was explicitly set in a function defined few lines above. Using `Expect` felt redundant.